WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -214,11 +214,50 @@ final class HubMenuViewModel: ObservableObject {
         }
     }
 
+    func updateDefaultConfigurationForPointOfSale(_ isPointOfSaleActive: Bool) {
+        updateTabBarVisibility(isPointOfSaleActive)
+        updateInAppNotifications(isPointOfSaleActive)
+    }
+
     deinit {
         NotificationCenter.default.removeObserver(self, name: .setUpTapToPayViewDidAppear, object: nil)
     }
 }
 
+// MARK: - Helper method for WooCommerce POS
+//
+private extension HubMenuViewModel {
+    // Hides the app's tab bars when Point of Sale is active
+    //
+    func updateTabBarVisibility(_ isPointOfSaleActive: Bool) {
+        guard let mainTabBarController = AppDelegate.shared.tabBarController else {
+            return
+        }
+        // If hidden, modifies the safeAreaInsets so that the view controller's view can occupy the space left by the hidden tab bar.
+        // Otherwise resets safeAreaInsets
+        mainTabBarController.tabBar.isHidden = isPointOfSaleActive
+
+        let bottomInset = 8.0
+        if mainTabBarController.tabBar.isHidden {
+            mainTabBarController.additionalSafeAreaInsets = UIEdgeInsets(top: 0, left: 0, bottom: bottomInset, right: 0)
+        } else {
+            mainTabBarController.additionalSafeAreaInsets = .zero
+        }
+
+        mainTabBarController.selectedViewController?.view.layoutIfNeeded()
```

Comment:
Yes, you're right.

Setting a non-zero `additionalSafeAreaInsets` triggers a layout update we need. Given we don't want to actually have a larger additional safe area inset, I think we could compromise on setting it to `.leastNonzeroMagnitude`.

Also, we don't need to call `layoutIfNeeded` anymore.

Given how hacky a solution is, we should leave a good comment explaining the solution. 

```suggestion
        mainTabBarController.additionalSafeAreaInsets.bottom = isPointOfSaleActive ? .leastNonzeroMagnitude : .zero
```

I think the best solution in the future could be presenting POS as a modal, and not pushing it into navigatation stack. In this case it should be presented above tab controller and we would avoid any issues such as this.

Reply:
I think presenting as a modal can cause issues with presenting other modals, and calling `dismiss` on the environment. We don't do that much, because of `posModal`, but we might in future use those things more...

Reply:
> Setting a non-zero additionalSafeAreaInsets triggers a layout update we need. 

That sounds good, thanks for digging into this 🙇 It does work the same with a non-zero value indeed. 

In https://github.com/woocommerce/woocommerce-ios/pull/13785/commits/a44f15cb0a71c46b50d8b89b8546f2ed054ec996 we update the inset to use `.leastNonzeroMagnitude`, removed the call to `layoutIfNeeded`, and added a longer comment with a reference to this issue.